xwiki-rendering-syntaxes/xwiki-rendering-syntax-xhtml/src/main/java/org/xwiki/rendering/internal/renderer/xhtml/link/DocumentXHTMLLinkTypeRenderer.java

```diff
@@ -90,6 +103,31 @@ protected String computeLabel(ResourceReference reference)
         return this.linkLabelGenerator.generate(reference);
     }
 
+    /**
+     * Implementation for computing a document link title when no title has been specified.
+     * Looks for a component implementing URITitleGenerator with a role hint matching the reference scheme.
+     * @param reference the reference of the link for which to compute the label
+     * @return the computed title
+     * @since 15.2RC1
+     */
+    private String computeCreateTitle(ResourceReference reference)
+    {
+        URITitleGenerator titleGenerator = this.defaultTitleGenerator;
+        String title;
+        if (this.componentManager.hasComponent(URITitleGenerator.class, reference.getType().getScheme())) {
+            try {
+                titleGenerator = this.componentManager.getInstance(URITitleGenerator.class,
+                 reference.getType().getScheme());
+            } catch (Exception e) {
+                LOGGER.error("Error while loading component for generating URI title: [{}]",
+                    ExceptionUtils.getRootCauseMessage(e));
+                LOGGER.debug("Full stack trace: ", e);
```

Comment:
See https://dev.xwiki.org/xwiki/bin/view/Community/CodeStyle/JavaCodeStyle/#HLoggingBestPractices for our rules regarding which logging level to use. I'm not sure failing to resolve a title generator is:

> an important problem that compromises the stability of the XWiki instance or that prevent an important system from working and that should not have happened

Reply:
Changing it to a warning :+1: